test/unitTests/core/test-seat.js

```diff
@@ -0,0 +1,135 @@
+import { test } from 'tape-promise/tape';
+import harden from '@agoric/harden';
+
+import { makeSeatMint } from '../../../core/seatMint';
+import { makeMint } from '../../../core/issuers';
+import { insist } from '../../../util/insist';
+
+/*
+ * A seat quantity may look like:
+ *
+ * {
+ *   id: {},
+ *   offerToBeMade: [rule1, rule2],
+ * }
+ *
+ * or:
+ *
+ * {
+ *   id: {},
+ *   offerMade: [rule1, rule2],
+ * }
+ *
+ */
+
+const allTrue = (prev, curr) => prev && curr;
+
+const ruleEqual = (leftRule, rightRule) => leftRule.rule === rightRule.rule;
+
+const amountEqual = (assay, leftRule, rightRule) =>
+  assay.equals(leftRule.amount, rightRule.amount);
+
+const offerEqual = (assays, leftOffer, rightOffer) => {
+  const isLengthEqual = leftOffer.length === rightOffer.length;
+  if (!isLengthEqual) {
+    return false;
+  }
+  return leftOffer
+    .map((leftRule, i) => {
+      return (
+        ruleEqual(leftRule, rightOffer[i]) &&
+        amountEqual(assays[i], leftRule, rightOffer[i])
+      );
+    })
+    .reduce(allTrue);
+};
+
+const setup = () => {
+  const moolaMint = makeMint('moola');
+  const simoleanMint = makeMint('simoleans');
+  const bucksMint = makeMint('bucks');
+
+  const moolaIssuer = moolaMint.getIssuer();
+  const simoleanIssuer = simoleanMint.getIssuer();
+  const bucksIssuer = bucksMint.getIssuer();
+
+  const moolaAssay = moolaIssuer.getAssay();
+  const simoleanAssay = simoleanIssuer.getAssay();
+  const bucksAssay = bucksIssuer.getAssay();
+
+  return harden({
+    mints: [moolaMint, simoleanMint, bucksMint],
+    issuers: [moolaIssuer, simoleanIssuer, bucksIssuer],
+    assays: [moolaAssay, simoleanAssay, bucksAssay],
+  });
+};
+
+test('seatMint', async t => {
+  const { assays } = setup();
+  const { seatMint, addUseObj } = makeSeatMint();
+
+  const makeUseObj = quantity => {
+    insist(quantity !== null)`the asset is empty or already used`;
+    if (quantity.offerToBeMade) {
+      return harden({
+        makeOffer: offer => {
+          insist(offerEqual(assays, offer, quantity.offerToBeMade));
+          // do things with the offer
+          return true;
+        },
+      });
+    }
+    if (quantity.offerMade) {
+      return harden({
+        claim: () => {
+          return [];
+        },
+      });
+    }
+    return harden({});
+  };
+
+  const purse1Quantity = harden({
+    id: harden({}),
```

Comment:
does the harden here add anything, since it's wrapped in an outer harden?

Reply:
It doesn't add anything technically, but I like hardening whenever I create a new object so that I don't forget it when it's actually important.